src/main/java/org/csanchez/jenkins/plugins/kubernetes/pipeline/ContainerExecDecorator.java

```diff
@@ -205,14 +222,23 @@ public void onClose(int i, String s) {
                         // The workspace is not known in advance, so we have to execute a cd command.
                         watch.getInput().write(
                                 String.format("cd \"%s\"%s", pwd, NEWLINE).getBytes(StandardCharsets.UTF_8));
+
                     }
 
                     if (environmentExpander != null) {
                         EnvVars envVars = new EnvVars();
                         environmentExpander.expand(envVars);
                         for (Map.Entry<String, String> entry : envVars.entrySet()) {
                             watch.getInput().write(
-                                    String.format("export %s=\"%s\"%s", entry.getKey(), entry.getValue(), NEWLINE).getBytes(StandardCharsets.UTF_8));
+                                    String.format("export %s=\"%s\"%s", entry.getKey(), StringEscapeUtils.escapeJavaScript(entry.getValue()), NEWLINE).getBytes(StandardCharsets.UTF_8));
+                        }
+                    }
+
+                    //setup specific command envs passed into cmd
+                    if (cmdEnvs != null) {
+                        for (String cmdEnv : cmdEnvs) {
+                            watch.getInput().write(
+                                    String.format("export \"%s\"%s", StringEscapeUtils.escapeJava(cmdEnv), NEWLINE).getBytes(StandardCharsets.UTF_8));
```

Comment:
This has the same problems as PR #218.

I suggest to solve both problems there, and close this one.

Reply:
Yeah - I had intended to drop that change from this PR or merge this one in after that, repulling to use the latest env setup.

Reply:
I merged this branch into your #218 branch to test locally.

Should I push that?

Reply:
One thing to note - the cmdenv variables come to us as a string, not a set of envVars. We might need to test it with breaking the value out and escaping it individually.

Reply:
Thanks, but there is an easy solution for that using the Jenkins API:

```
                    EnvVars envVars = new EnvVars();

                    if (cmdEnvs != null) {
                        for (String cmdEnv : cmdEnvs) {
                            envVars.addLine(cmdEnv);
                        }
                    }
```

Just checking how I can test that right now, as `cmdEnvs` has been empty for all my previous manual tests..

Reply:
Rerunning the existing test @carlossg provided should confirm it works just as well.

Reply:
I'll also open a new PR once the environment variables change set is merged and close this one.

Reply:
> I'll also open a new PR once the environment variables change set is merged and close this one.

Sounds like a better idea then merging this PR into #218.

When you do the merge, please note that you can just do

```java
                    EnvVars envVars = new EnvVars();
                    if (environmentExpander != null) {
                        environmentExpander.expand(envVars);
                    }

                    //setup specific command envs passed into cmd
                    if (cmdEnvs != null) {
                        for (String cmdEnv : cmdEnvs) {
                            envVars.addLine(cmdEnv);
                        }
                    }

                    this.setupEnvironmentVariable(envVars, watch);
```

Reply:
Will do - I'll close this one out and make those changes.